test/bot-events.test.js

```diff
@@ -141,4 +141,47 @@ describe('Bot Events', function() {
     secondCall.args[0].should.equal('Channel not found in config, not joining:');
     secondCall.args[1].should.equal(channel);
   });
+
+  it('should send joins to slack', function() {
+    var a = this.bot.nickname;
```

Comment:
The max line length is there to make it easier to use split windows (without having to scroll horizontally). Both string concatenation and function calls can be broken up into multiple lines in JS, so the long lines could be changed into something like:

```javascript
var msg = nick + 
  ' has quit the ' + b[i] 
  + ' channel';

Bot.prototype.sendToSlack
  .should.have.been
  .calledWithExactly(d, b[i], msg);
```